### PR TITLE
chore(cardinal): clean up message stuff 

### PR DESCRIPTION
**Pull request by technicallyty**
## Overview

while working on 970, i found some quirky stuff going on in the message API and some tests so im opening a small PR to clean it up.

## Brief Changelog

- remove redundant `name` argument from `msgManager.RegisterMessage`
- remove duplicate `NewMessageType` from testutils, can just use the regular `NewMessageType`
- remove `SetName` and `SetGroup` from `MessageType`
- moved `GetMessage` to testutils.go to reduce file clutter
- tidied all go.mods 
- re-generated all gen stuff

### CODE REVIEW

File: cardinal/testutils/testutils.go [hunk 1]
```diff
@@ -85,3 +89,18 @@ func AddTransactionToWorldByAnyTransaction(
 
 	_, _ = world.AddTransaction(txID, value, tx)
 }
+
+func GetMessage[In any, Out any](wCtx engine.Context) (*message.MessageType[In, Out], error) {
+	var msg message.MessageType[In, Out]
+	msgType := reflect.TypeOf(msg)
+	tempRes, ok := wCtx.GetMessageByType(msgType)
+	if !ok {
+		return nil, eris.Errorf("Could not find %s, Message may not be registered.", msg.Name())
```
Comment (smsunarto) [inline cardinal/testutils/testutils.go:98]: ser, i thought you are a %q maxi

**Review by smsunarto — APPROVED** [review]
lgtm

File: cardinal/testutils/testutils.go (hunk 1 above)
Comment (technicallyty) [reply cardinal/testutils/testutils.go:98]: this was just a copy pasta, i didnt write this

Comment (technicallyty) [reply cardinal/testutils/testutils.go:98]: can ninja it though

**Pull request merged**